tree/tree/src/TChain.cxx

```diff
@@ -804,12 +805,17 @@ Long64_t TChain::Draw(const char* varexp, const char* selection,
 
 TBranch* TChain::FindBranch(const char* branchname)
 {
+   std::string name{branchname};
```

Comment:
Can you add a code comment explaining the net/resulting effect of conditionally stripping the chain's name from the branch name?